model/Image.php

```diff
@@ -163,8 +163,35 @@ public function getTag() {
 	public function forTemplate() {
 		return $this->getTag();
 	}
+	
+	/**
+	 * Gets the source image URL for this resource
+	 * 
+	 * @return string
+	 */
+	public function getSourceURL() {
+		return parent::getURL();
+	}
 
 	/**
+	 * Gets the relative URL accessible through the web. If forced resampling is enabled 
+	 * the URL will point to an optimised file, if it is smaller than the original
+	 *
+	 * @uses Director::baseURL()
+	 * @return string
+	 */
+	public function getURL() {
+		if ($this->config()->force_resample) {
+			//return $resampled->getURL();
+			$resampled = $this->getFormattedImage('Resampled');
+			if ($resampled->getAbsoluteSize() < $this->getAbsoluteSize()) {
```

Comment:
Yeah I guess that I still expect a resampled image though, not the original? Not saying that its a bad thing, ideally what ever is served should be the smallest option - I just think that the expectation here is for a resampled image to be returned if its resampling ;)